tidy3d/components/data/data_array.py

```diff
@@ -283,6 +285,79 @@ def multiply_at(self, value: complex, coord_name: str, indices: List[int]) -> Da
         self_mult[{coord_name: indices}] *= value
         return self_mult
 
+    def interp(
+        self,
+        coords: Union[Mapping[Any, Any], None] = None,
+        method: InterpOptions = "linear",
+        assume_sorted: bool = False,
+        kwargs: Union[Mapping[str, Any], None] = None,
+        **coords_kwargs: Any,
+    ):
+        """Interpolate this DataArray to new coordinate values.
+
+        Parameters
+        ----------
+        coords : Union[Mapping[Any, Any], None] = None
+            A mapping from dimension names to new coordinate labels.
+        method : InterpOptions = "linear"
+            The interpolation method to use.
+        assume_sorted : bool = False
+            If True, skip sorting of coordinates.
+        kwargs : Union[Mapping[str, Any], None] = None
+            Additional keyword arguments to pass to the interpolation function.
+        **coords_kwargs : Any
+            The keyword arguments form of coords.
+
+        Returns
+        -------
+        DataArray
+            A new DataArray with interpolated values.
+
+        Raises
+        ------
+        KeyError
+            If any of the specified coordinates are not in the DataArray.
+        """
+        if self.tracers is not None:  # use custom interp if using traced data
+            coords = either_dict_or_kwargs(coords, coords_kwargs, "interp")
+
+            missing_keys = set(coords) - set(self.coords)
+            if missing_keys:
+                raise KeyError(f"Cannot interpolate: {missing_keys} not in coords.")
+
+            obj = self if assume_sorted else self.sortby(list(coords.keys()))
+
+            points = tuple(dict(obj.coords).values())
+            xi = tuple(coords.get(k, obj.coords[k]) for k in obj.dims)
+            vals = interpn(points, obj.tracers, xi, method=method)
+
+            da = DataArray(vals, dict(obj.coords) | coords)  # tracers go into .attrs
+            if isbox(self.values.flat[0]):  # if tracing .values instead of .attrs
```

Comment:
or `da.contains_tracers`? although this seems to no longer exist?

Reply:
maybe it's fine for now until we standardize all of this.

Reply:
Yeah I can also reintroduce that property, but I think the whole handling needs some work so maybe we leave it as is for the patch.

Reply:
ok